src/composables/protocol.ts

```diff
@@ -0,0 +1,56 @@
+import type {
+  Class,
+  IDefaultComposableOptions,
+  Protocol,
+} from '@/types';
+import { BaseProtocol } from '@/protocols/BaseProtocol';
+import { AeternityProtocol } from '@/protocols/aeternity/libs/AeternityProtocol';
+import { BitcoinProtocol } from '@/protocols/bitcoin/libs/BitcoinProtocol';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+} from '@/constants';
+import { watchUntilTruthy } from '@/utils';
+import { useAccounts } from './accounts';
+
+let isProtocolRegistryReady = true;
+
+const protocolClassReferences: Record<Protocol, Class<BaseProtocol> > = {
+  [PROTOCOL_AETERNITY]: AeternityProtocol,
+  [PROTOCOL_BITCOIN]: BitcoinProtocol,
+};
+const protocolInstanceRegistry: Partial<Record<Protocol, BaseProtocol>> = {};
+
+export function useProtocol({ store }: IDefaultComposableOptions) {
+  const { activeAccount } = useAccounts({ store });
+
+  function registerProtocol(protocol: Protocol, instance: BaseProtocol) {
+    if (protocolInstanceRegistry[protocol]) {
+      throw new Error('Protocol instance already exists in the registry');
+    }
+    protocolInstanceRegistry[protocol] = instance;
+  }
+
+  function getProtocolFromRegistry(protocol: Protocol): BaseProtocol | undefined {
+    return protocolInstanceRegistry[protocol];
+  }
+
+  /**
+   * get SDK of given protocol or active account protocol
+   */
+  async function getProtocolInstance(targetProtocol?: Protocol): Promise<BaseProtocol> {
```

Comment:
Why `async`? The Base class has no constructor at all so instantiation is synchronous.

Reply:
Yes but we https://github.com/aeternity/superhero-wallet/blob/48716742e58d2426d8b03743da7a85c4c494a502/src/composables/protocol.ts#L44

If not then when we load the app many components would call `getProtocolInstance` at once and we might end up creating multiple instances of the class. 

I can keep it very simple but it would create a class instance every time but that's fine I guess as it's stateless.
```js
 function getProtocolInstance(targetProtocol?: Protocol): BaseProtocol {
    const protocol = targetProtocol || activeAccount.value.protocol;
    return new (protocolClassReferences[protocol])();
  }
```

Reply:
Yeah, but it's totally wrong. The `watchUntilTruthy` requires reactive property. `isProtocolRegistryReady` is not watchable. The code you created works only because the class instantiation is immediate.

Reply:
> I can keep it very simple but it would create a class instance every time
No, it won't because the instantiation is immediate and the registry is filled with the instance immediate after calling the `getProtocolInstance` :)

Reply:
Kept it very simple. Please have a fresh look https://github.com/aeternity/superhero-wallet/pull/2193/files#diff-b04f42206a6a2e9b4688bc42b5193f5cf351c98c98f28610bc7f842a6e07489b
@peronczyk